server/etcdserver/server.go

```diff
@@ -892,6 +895,31 @@ func (s *EtcdServer) revokeExpiredLeases(leases []*lease.Lease) {
 	})
 }
 
+// ensureLeadership ensures current member is still the leader before
+// performing the leases revoking operations. It can mitigate the issue
+// https://github.com/etcd-io/etcd/issues/15247, but it can't completely
+// resolve it because the leases may be wrongly revoked by the new leader.
+// Refer to the document in
+// https://github.com/etcd-io/etcd/issues/15247#issuecomment-1777862093.
+func (s *EtcdServer) ensureLeadership() bool {
+	lg := s.Logger()
+	if err := s.linearizableReadNotify(s.ctx); err != nil {
+		lg.Warn("Ignore leases revoking request due to failing to ensure current member's leadership",
+			zap.Error(err))
```

Comment:
log is related to lease revoke but `ensureLeadership` can be used by any etcd internal process instead of just lease revoking.

Reply:
Good point. Updated.